lib/codegen/meta/gen_instr.py

```diff
@@ -279,17 +279,29 @@ def gen_instruction_data_impl(fmt):
                         args_eq = 'args1 == args2'
                     for field in f.imm_fields:
                         members.append(field.member)
+
                     pat1 = ', '.join('{}: ref {}1'.format(x, x)
                                      for x in members)
                     pat2 = ', '.join('{}: ref {}2'.format(x, x)
                                      for x in members)
+
                     with fmt.indented('({} {{ {} }}, {} {{ {} }}) => {{'
                                       .format(n, pat1, n, pat2), '}'):
-                        fmt.line('opcode1 == opcode2 &&')
-                        for field in f.imm_fields:
-                            fmt.line('{}1 == {}2 &&'
-                                     .format(field.member, field.member))
-                        fmt.line(args_eq)
+                        line = 'opcode1 == opcode2'
+                        if len(f.imm_fields) or args_eq is not None:
+                            line += ' &&'
+                        fmt.line(line)
+
+                        for (idx, field) in enumerate(f.imm_fields):
+                            line = ('{}1 == {}2'
+                                    .format(field.member, field.member))
+                            if (idx != len(f.imm_fields) - 1 or args_eq is not
+                                    None):
+                                line += ' &&'
+                            fmt.line(line)
+
+                        if args_eq is not None:
+                            fmt.line(args_eq)
```

Comment:
Could this code be simplified by moving the '&&'s to the following lines? Eg.:
```
     opcode1 == opcode2
     && field == field
     && field == field
     && /* args_eq */
```
That way, each additional line could add its own '&&', rather than having to check whether one will be needed.

Also, this follows recent versions of rustfmt which have been [formatting code this way](https://github.com/cretonne/cretonne/blob/master/lib/codegen/src/isa/x86/abi.rs#L468).